server/athenian/api/controllers/miners/github/contributors.py

```diff
@@ -21,14 +21,42 @@ async def mine_contributors(repos: Collection[str],
                             time_to: Optional[datetime],
                             db: databases.Database,
                             cache: Optional[aiomcache.Client],
-                            with_stats: Optional[bool] = True) -> List[dict]:
+                            with_stats: Optional[bool] = True,
+                            with_roles: List[str] = None) -> List[dict]:
     """Discover developers who made any important action in the given repositories and \
     in the given time frame."""
     time_from = time_from or datetime(1970, 1, 1, tzinfo=timezone.utc)
     now = datetime.now(timezone.utc) + timedelta(days=1)
     time_to = time_to or datetime(now.year, now.month, now.day, tzinfo=timezone.utc)
 
-    return await _mine_contributors(repos, time_from, time_to, with_stats, db, cache)
+    if with_roles:
+        fetch_stats = True
+    else:
+        fetch_stats = with_stats
+
+    contribs = await _mine_contributors(repos, time_from, time_to, fetch_stats, db, cache)
+    if not with_roles:
+        return contribs
+
+    # Doing this in post-processing to avoid decreasing cache hits. Otherwise we could
+    # consider caching the different `fetch_*` functions indidually as a inner caching
+    # layer if this becomes a bottleneck.
+    filtered_contribs = []
+    # We could get rid of these re-mapping, maybe worth looking at it along with the
+    # definition of `DeveloperUpdates`
+    role_mapping = {"author": "prs"}
```

Comment:
Let's hardcode this for now, I don't expect any other deviations between the models.

Reply:
What do you mean exactly with hardcoding it?  You mean something like:
```
        has_active_role = sum(c["stats"].get(role if role != "author" else "prs", 0)
                              for role in as_roles) > 0
```
it seems less readable to me.

Reply:
This is both faster and more obvious:
```python
for contrib in contribs:
    role_stats = contrib["stats"]
    has_active_role = False
    for role in as_roles:
        if role == "author":
            role = "prs"
        if role_stats.get(role, 0) > 0:
            has_active_role = True
            break
```

Reply:
Just pushed, I've done something similar.

Reply:
Sorry just pushed a simplification [here](https://github.com/athenianco/athenian-api/pull/447/commits/3cffae66994eab4d416ff1fd515e588dd11d91c0) just in case you started reviewing in the meantime.